### ISSUE TITLE
Use pattern matching (ID0078) false positive when using .editorconfig

### ISSUE THREAD

**Issue by FlorianObermayer** [dotnet/roslyn#52080]
### Analyzer

**Diagnostic ID**: [ID0078](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0078): `Use pattern matching`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.201](https://dotnet.microsoft.com/download/dotnet/5.0)

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

We came across multiple conditions that are incorrectly flagged for not using pattern matching. The resulting refactor usually breaks code or sometimes changes the logic of the expression.

It only occurs when using a `.editorconfig` with this line: 
`dotnet_analyzer_diagnostic.category-Style.severity = warning`

##### Non-compiling:
`if (args.Text.Length == 1 && Convert.ToChar(args.Text) != escapeKey)`
→
`if (Convert.ToChar(args.Text) is 1 and not (char)escapeKey)` 

`if (physicalPath.Length > 1 && physicalPath[1] == ':')`
→
`if (physicalPath[1] is > 1 and ':')`

##### Compiling but with altered logic:
`if (handler != null && ValidationHost != null)`
→
`if (ValidationHost is not null and not null)` 

### Steps To Reproduce

I created a minimal example to reproduce one of the false-positives:

`Program.cs`
```csharp
public class Program
{
  public static void Main()
  {
    string str = string.Empty;
    if (str.Length > 1 && str[1] == char.MinValue)
    {
      // do something
    }
  }
}
```

`.editorconfig`
```editorconfig
root = true
[*.{cs,vb}]
dotnet_analyzer_diagnostic.category-Style.severity = warning
```

Note that I could **not** reproduce the other issues outside our production environment (which is a WPF app migrated from .Net Framework 4.7.2 to .NET 5).

**Comment by jinujoseph** [comment]
cc @CyrusNajmabadi
